File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.ts
function uniformArray(nums: number[]): boolean {
    let mn = Infinity;
    let oddCnt = 0;

    for (const x of nums) {
        if (x < mn) mn = x;
        if (x % 2 === 1) oddCnt++;
    }

    return (mn % 2 === 1) || (oddCnt === 0);
};

File: 3876-construct-uniform-parity-array-ii/3876-construct-uniform-parity-array-ii.rs
impl Solution {
    pub fn uniform_array(nums1: Vec<i32>) -> bool {
        let mut mn = i32::MAX;
        let mut odd_cnt = 0;

        for &x in nums1.iter() {   // iterate over reference to avoid moving
            if x < mn { mn = x; }
            if x % 2 == 1 { odd_cnt += 1; }
        }

        (mn % 2 == 1) || (odd_cnt == 0)
    }
}